stacks-signer/src/signer.rs

```diff
@@ -1438,6 +1480,42 @@ impl Signer {
         Ok(())
     }
 
+    /// Send DKG results to the miner coordinator so that it can coordinate signing rounds
+    fn send_dkg_results(&mut self, dkg_public_key: &Point) {
+        let Some(dkg_results_bytes) = self.pending_dkg_results.remove(dkg_public_key) else {
+            warn!("{self}: No DKG results found for key: {dkg_public_key}");
+            return;
+        };
+
+        if let Err(e) = self
+            .stackerdb
+            .send_message_bytes_with_retry(&MessageSlotID::DkgResults, dkg_results_bytes)
+        {
+            error!("{}: Failed to send DKGResults message to StackerDB, will continue operating.", self.signer_id;
+                       "error" => %e);
+        };
+    }
+
+    /// Prepare the DKG results message to be sent once the public key is approved
+    fn prepare_dkg_results(&mut self, dkg_public_key: Point) {
+        let mut dkg_results_bytes = vec![];
+        debug!(
+            "{self}: DKG public key changed. Sending DKG results";
+            "dkg_public_key" => %dkg_public_key
+        );
+        if let Err(e) = SignerMessage::serialize_dkg_result(
+            &mut dkg_results_bytes,
+            &dkg_public_key,
+            self.coordinator.party_polynomials.iter(),
+        ) {
+            error!("{}: Failed to serialize DKGResults message for StackerDB, will continue operating.", self.signer_id;
+                   "error" => %e);
```

Comment:
Shouldn't this return an error instead of soldering on and trying to insert the (presumably undefined?) `dkg_results_bytes` into `pending_dkg_results`?

Reply:
Yeah, I'll add error handling to this function and the `process_dkg` function which calls it.

Reply:
Done. This does change the logic as you noted since the above function would continue on errors previously. I haven't observed any test failures due to this yet, but I don't know the signer well enough to guarantee that this is 100% safe - even though it feels like it should be better to return errors eagerly.